### ISSUE TITLE
consulCatalog issue when serviceName contains a dot

### ISSUE THREAD

**Issue by joonas-fi** [containous/traefik#475]
tl;dr: I only got `consulCatalog` to work by adding `--consulCatalog.constraints=tag==foo` to `argv` and adding `traefik.tags=traefik` tag to service definition in Consul - which sounds very artificial since according to docs the defaults should work with every Consul service and no constraints would be required. Without the constraints Traefik fails with very unhelpful error message.

Version: `traefik:latest` (today is 2016-06-21)

----------------

Full description:

`constraints` is documented as optional. BUT, running without it yields:

	time="2016-06-21T10:19:31Z" level=error msg="Error loading new configuration, aborted Undefined backend: " 

This error message just leaves me dumbfounded. There's nothing in it that helps me diagnose what's wrong. Desperately I'll try to add a `constraint` to filter Consul-advertised services to only one service that has very explicit tags and so it should work - at least for drilling down on the issue.

When I try to apply a constraint using the documented way:

	constraints = ["tag==foo"]

This yields:

	Error reading TOML config file /etc/traefik/traefik.toml : Type mismatch for 'main.TraefikConfiguration.ConsulCatalog': Type mismatch for 'provider.ConsulCatalog.Constraints': Type mismatch for types.Constraint. Expected map but found 'string'.

This also doesn't work:

	constraints = "tag==foo"

Nor:

	[consulCatalog.constraints]
	tag = "foo"

I got it working only after I added `--consulCatalog.constraints=tag==foo` to `argv`.

To this moment, I don't know why I have to add a constraint, since according to docs there doesn't need to be any Consul tags with a service for it to work correctly in Traefik (only tag that seems really important is `traefik.frontend.rule` but it has a sensible default anyways (defining `domain` didn't help).

Also, it is super confusing to specify constraint `tag==foo` and to realize it doesn't look at **Consul tag** "foo", but **Traefik tag** foo instead, which is **implemented** with Consul tags, but with a prefix (`traefik.tags=foo`). I think I might just have a stroke dealing with all this complexity. :D

**Comment by emilevauge** [comment]
Ping @samber

**Comment by joonas-fi** [comment]
Umm this might be something else, as now Traefik stopped working with consulCatalog even though I have the described constraint system in effect.

I'm still researching this, but currently this seems to trigger when I bring a service online, whose `service name` in consul contains a dot (`.`).

**Comment by joonas-fi** [comment]
Yeah, I've tested many times stopping and starting my only service whose name contains a dot, and it appears consistent that the log line emits when that service comes online:

	Error loading new configuration, aborted Undefined backend: 

When the dot-containing service is present in Consul, after restarting Traefik the `consul_catalog` tab in the UI does not pop up (even though I have another service, not containing a dot, in Consul that should appear in Traefik).

The second I stop my dot-containing service `consul_catalog` shows up in Traefik. If that dot-containing service comes online when Traefik is running, the other valid service shows up under `consul_catalog` tab, but it seems that updates are not synced from `consul_catalog` as long as the dot-containing service name exists.

Workaround until now: I'll make the service not contain a dot.

**Comment by joonas-fi** [comment]
P.s. `constraints = ["tag==foo"]` not working is still a bug, right?

**Comment by emilevauge** [comment]
@joonas-fi 
>P.s. constraints = ["tag==foo"] not working is still a bug, right?
The documentation is false:
https://github.com/containous/traefik/issues/451#issuecomment-227387522
